rat/compiler.py

```diff
@@ -390,7 +390,7 @@ def compile(data_df: pandas.DataFrame, parsed_lines: List[ops.Expr]):
                     index_use_identifier = (op.index.get_key(), op.index.shifts)
 
                     df_index = (
-                        op.index.get_key() if isinstance(lhs, ops.Data) else lhs.variable.index.check_and_return_index(op.index.get_key())
+                        op.index.get_key()  # if isinstance(lhs, ops.Data) else lhs.variable.index.check_and_return_index(op.index.get_key())
```

Comment:
Here's the weirdness -- I think if you uncomment this and put the code back you'll get a compiler error

Reply:
#46 I mean

Reply:
Hmm and commenting that still makes the tests pass. Looks like `check_and_return_index` is definitely sus.